src/main/java/com/google/step/youtube/YouTubeInfoScraper.java

```diff
@@ -103,6 +105,25 @@ public YouTubeInfoScraper() {
         return Optional.of(response.getItems());
     }
 
+    /**
+     * @param keyword Word to search with.
+     * @return an optional list of videoIds.The optional will be empty if id is
+     *         invalid.
+     */
+    public Optional<List<String>> scrapeVideoIdsFromSearch(String keyword) throws IOException {
+        SearchListResponse response = youTubeClient.search().list("").setMaxResults(50L).setQ(keyword)
+                .setFields("items(id)").execute();
+        if (response.getItems() == null) {
+            return Optional.empty();
+        }
+        checkState(!response.getItems().isEmpty(), "Expected more than 0 SearchResult items to be found.");
```

Comment:
What if the query is a keyboard smash and there are no results found? We should probably still return `Optional.empty()` in this case.

Reply:
The API will return null if empty.  We are just checking for impossible scenarios.